homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -573,12 +579,18 @@ def musiccast_zone_entity(self) -> MusicCastMediaPlayer:
 
         return self
 
-    async def update_all_mc_entities(self):
+    async def update_all_mc_entities(self, check_clients=False):
         """Update the whole musiccast system when group data change."""
-        for entity in self.get_all_mc_entities():
-            if entity.is_server:
+        # First update all servers as they provide the group information for their clients
+        for entity in self.get_all_server_entities():
+            if check_clients or self.coordinator.musiccast.group_reduce_by_source:
                 await entity.async_check_client_list()
-            entity.async_write_ha_state()
+            else:
+                entity.async_write_ha_state()
+        # Then update all other entities
+        for entity in self.get_all_mc_entities():
+            if not entity.is_server:
+                entity.async_write_ha_state()
```

Comment:
For a future PR: It would be better to use our dispatch helper to signal these entities to update etc themselves if they are not a server etc.